bluemira/optimisation/_geometry/_tools.py

```diff
@@ -102,17 +116,13 @@ def calculate_signed_distance(
     return signed_distance_2D_polygon(s.T, zone_points.T).T
```

Comment:
Might be worth exposing the dl, byedges, ndiscr in the shape discretisation as well... Just thinking about a "stable" API here; with this discretisation control on the KOZ, we might also want the same level of control on the shape. If we can do that in the same way, API-wise, it would be good. Not sure if this means a different interface, but it could well do.

Reply:
Ah yes, that's true... It's likely only to be the case for very low ndiscr, or polygons with many many sides. But it's a potential problem yes. OK, anyway, this PR looks good but I'd like to have a play with it tomorrow and get back to you.

Reply:
I've added the ability to set `n_shape_discr`